app/vmagent/remotewrite/client.go

```diff
@@ -433,6 +473,7 @@ again:
 		c.retriesCount.Inc()
 		goto again
 	}
+	c.upgradeVMProto(resp)
```

Comment:
Question

Should we handle a case where a request failed because zstd compression was used, but a server does not support it ?

Reply:
yes, we should. prometheus can return 415 status

https://github.com/prometheus/prometheus/blob/c88d0b0e0a4a1c4281e04bc68c86bd37a950712f/storage/remote/write_handler.go#L157